Add config for Gallery study_short_name.

will-moore opened the pull request:
1 more config from https://github.com/ome/omero-gallery/pull/42

This regex uses the Study ```Name``` (e.g. ```idr0010-doil-dnadamage/screenA```) and produces e.g. ```idr0010A```.
See query at https://github.com/ome/omero-gallery/pull/42/files#r299405852
As the code might change there, best to wait for that PR to be merged before this one.

will-moore commented:
Travis fails with ```"psql:omero-20190702-102741-756930.sql:9478: ERROR:  relation \"i_logicalchannel_emissionwave\" already exists"]```

manics commented:
That's the postgres problem, https://github.com/ome/ansible-role-postgresql/pull/15 could be used to pin the version

will-moore commented:
Added support for 'template' at https://github.com/ome/omero-gallery/pull/42/commits/62c7db52a753b69985942bcd7ac2255a1fa48c51 and updated the config for that in 8acdc17

will-moore commented:
@joshmoore @manics Is it possible to get this deployed onto testing server where we can test it along with https://github.com/ome/omero-gallery/pull/42 (without merging either just yet)?
Thanks.

manics commented:
Could merge ome/omero-gallery#42  into a develop branch and make a devel tag?

will-moore commented:
@manics I can manually deploy the scc merged branch onto the testing server so I don't need any help with https://github.com/ome/omero-gallery/pull/42.
But I do need help with *this* PR to get the settings in place before I update the gallery app.

sbesson commented:
This has been deployed to `test64`

will-moore commented:
Seems that /gallery_settings/ JSON is 
```
"STUDY_SHORT_NAME": [
{
  "key": "Name"
},
{
   "regex": "^(.*?)-.*?(.)$"
},
{
   "template": "12"
}
],
```
when what we want is a single dict in a list and no escaping!

```
"STUDY_SHORT_NAME": [
{
"key": "Name",
"regex": "^(.*?)-.*?(.)$"
"template": "{{1}}{{2}}"
}]
```

manics commented:
https://travis-ci.org/IDR/deployment/jobs/557902967
```
    TASK [Gathering Facts] *********************************************************
ERROR! Syntax Error while loading YAML.
  found unknown escape character
The error appears to have been in '/home/travis/build/IDR/deployment/ansible/group_vars/omero-hosts.yml': line 510, column 16, but may
be elsewhere in the file depending on the exact syntax problem.
The offending line appears to be:
    regex: "^(.*?)-.*?(.)$"
    template: "\{\{1}}\{\{2}}"
               ^ here
    
ERROR: 
```
My best guess: try `\\` instead of `\`

will-moore commented:
@sbesson Travis is happy now. Could you give this a try? Thx.

will-moore commented:
We seem to need escaping of the new template AND fixing of indenting to make sure we have a single dict (with 3 keys) as the only item in the list

JSON is now:

```
"STUDY_SHORT_NAME": [
  {
    "key": "Name"
  },
  {
    "regex": "^(.*?)-.*?(.)$"
  },
  {
    "template": "12"
  }
],
```

will-moore commented:
@manics Could you redeploy these latest settings on prod70, for testing fixes to https://github.com/ome/omero-gallery/pull/42 (which are already deployed there). Thanks.

manics commented:
Deployed to `test70` (I presume that's what you meant):
```diff
-config set -- omero.web.gallery.title 'Welcome to IDR'
+config set -- omero.web.gallery.title 'IDR: Image Data Resource'
+config set -- omero.web.gallery.heading 'Welcome to IDR'
```
Note since I used the Ansible playbook it may have reverted gallery to the current tag `3.2.0a7`, if so you'll need to redeploy the dev build

manics commented:
After https://github.com/ome/omero-gallery/pull/42 is released could you:
- include the release version
- change this to `https`: https://github.com/IDR/deployment/blob/3866f31fdf1391c951e57ca80585ab3ef1e4a32f/ansible/group_vars/omero-hosts.yml#L413

manics commented:
Deployed to `test70`

manics commented:
Deployed to `prod70`
